test/framework/functional/output_http.go

```diff
@@ -54,25 +69,62 @@ codec = "ndjson"
 `
 )
 
+func VectorConfFactory(profile configv1.TLSProfileType) string {
+	minTLS := ""
+	ciphers := ""
+	if profile != "" {
+		if spec, found := configv1.TLSProfiles[profile]; found {
+			minTLS = string(spec.MinTLSVersion)
+			ciphers = strings.Join(spec.Ciphers, ",")
+		}
+
+	}
+	tmpl, err := template.New("").Parse(VectorHttpSourceConfTemplate)
+	if err != nil {
+		log.V(0).Error(err, "Unable to parse the vector http conf template")
+		os.Exit(1)
+	}
+	b := &bytes.Buffer{}
+	if err := tmpl.ExecuteTemplate(b, "", struct {
+		MinTLS  string
+		Ciphers string
+	}{
+		MinTLS:  minTLS,
+		Ciphers: ciphers,
+	}); err != nil {
+		log.V(0).Error(err, "Unable execute vector http conf template")
+		os.Exit(1)
+	}
+	return b.String()
+}
+
 func (f *CollectorFunctionalFramework) AddVectorHttpOutput(b *runtime.PodBuilder, output logging.OutputSpec) error {
+	return f.AddVectorHttpOutputWithConfig(b, output, "", nil)
+}
+
+func (f *CollectorFunctionalFramework) AddVectorHttpOutputWithConfig(b *runtime.PodBuilder, output logging.OutputSpec, toml configv1.TLSProfileType, secret *corev1.Secret) error {
```

Comment:
the toml name is confusing here, how about tlsProfile?

Reply:
:+1: that is left over from a previous version